framework/src/preference/shared_preference.c

```diff
@@ -0,0 +1,295 @@
+/****************************************************************************
```

Comment:
Almost same as private_preference.c.
Now we learned from SAM that it is not good choice..